gwpy/table/io/gravityspy.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) Scott Coughlin (2017)
+#
+# This file is part of GWpy.
+#
+# GWpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# GWpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with GWpy.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Input/output methods for tabular data.
+
+Access to GravitySpy and O1GlitchClassification triggers requires access
+to a PostgresSQL database. Users can set the username and password for
+connections in the following environment variables
+
+- ``QUEST_SQL_USER``
+- ``QUEST_SQL_PASSWORD``
+
+These can be found https://secrets.ligo.org/secrets/144/. The description
+is the username and thesecret is the password.
+"""
+import os
+
+from .. import GravitySpyTable
+from .. import EventTable
+from astropy.table import Table
+from .fetch import register_fetcher
+
+__author__ = 'Scott Coughlin <scott.coughlin@ligo.org>'
+
+# get GravitySpy environment variables
+QUEST_SQL_USER = os.getenv('QUEST_SQL_USER', None)
+QUEST_SQL_PASSWORD = os.getenv('QUEST_SQL_PASSWORD', None)
+
+
+def get_gravityspy_triggers(tablename, selection=None, **kwargs):
+    """Fetch data into an `GravitySpyTable`
+
+    Parameters
+    ----------
+    table : `str`,
+        The name of table you are attempting to receive triggers
+        from.
+
+    selection
+        other filters you would like to supply
+        underlying reader method for the given format
+
+    .. note::
+
+       For now it will attempt to automatically connect you
+       to a specific DB. In the future, this may be an input
+       argument.
+
+    Returns
+    -------
+    table : `GravitySpyTable`
+
+    Notes"""
+    try:
+        import pandas as pd
+    except ImportError as e:
+        e.args = ('pandas is required to download triggers',)
+        raise
+
+    try:
+        from sqlalchemy.engine import create_engine
+    except ImportError as e:
+        e.args = ('sqlalchemy is required to download triggers',)
+        raise
+
+    if (not QUEST_SQL_USER) or (not QUEST_SQL_PASSWORD):
```

Comment:
@scottcoughlin2014, it makes more sense to me to access these variables directly from `os.environ` at run time (rather than when the module gets imported), so that users can set them whenever they like. Also, does it not make sense to add `user` and `password` keyword arguments so they can be provided directly to the method?

Also, why are these variables called `QUEST_SQL_xxx`, does it make sense at this time to rename them `GRAVITYSPY_SQL_USER` and `GRAVITYSPY_SQL_PASSWORD`, at least for the GWpy interface?